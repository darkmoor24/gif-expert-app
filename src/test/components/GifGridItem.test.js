import React from "react";
// import { render } from "@testing-library/react";
import "@testing-library/jest-dom";
import { GifGridItem } from "../../components/GifGridItem";
import { shallow } from "enzyme";


describe('Probando el componente < GifGridItem />', () => {
    const title = "Probando el componente";
    const url = "https://localhost/algo.jpg";
    const wrapper = shallow(<GifGridItem title={ title } url={ url }/>);

    test('Debe mostrar el componente correctamente', () => { 
        expect(wrapper).toMatchSnapshot();
    });

    test('Debe de tener un <p></> con el title', () => {
        const p = wrapper.find("p");

        expect(p.text().trim()).toBe(title);
    });

    test('Debe tener un <img /> con los props', () => {
        const img = wrapper.find("img");

        expect(img.prop('alt')).toBe(title);

        expect(img.prop('src')).toBe(url);
    });

    test('Debe de tener clase card', () => {
        const div = wrapper.find("div");

        expect(div.hasClass("card")).toBe(true);
    });
});