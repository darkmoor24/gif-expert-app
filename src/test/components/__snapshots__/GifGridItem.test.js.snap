// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Probando el componente < GifGridItem /> Debe mostrar el componente correctamente 1`] = `
<div
  className="card"
>
  <img
    alt="Probando el componente"
    src="https://localhost/algo.jpg"
  />
  <p>
    Probando el componente
  </p>
</div>
`;
